src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs

```diff
@@ -19,24 +19,31 @@ internal sealed class SessionWithSolution : IDisposable
         private readonly RemoteHostClient.Connection _connection;
         private readonly PinnedRemotableDataScope _scope;
 
-        public static async Task<SessionWithSolution> CreateAsync(RemoteHostClient.Connection connection, PinnedRemotableDataScope scope, CancellationToken cancellationToken)
+        public static async Task<SessionWithSolution> CreateAsync(RemoteHostClient.Connection connection, Solution solution, CancellationToken cancellationToken)
         {
-            var sessionWithSolution = new SessionWithSolution(connection, scope);
+            Contract.ThrowIfNull(connection);
+            Contract.ThrowIfNull(solution);
 
+            PinnedRemotableDataScope scope = null;
             try
             {
+                scope = await solution.GetPinnedScopeAsync(cancellationToken).ConfigureAwait(false);
+
                 // set connection state for this session.
                 // we might remove this in future. see https://github.com/dotnet/roslyn/issues/24836
                 await connection.InvokeAsync(
                     WellKnownServiceHubServices.ServiceHubServiceBase_Initialize,
                     new object[] { scope.SolutionInfo },
                     cancellationToken).ConfigureAwait(false);
 
-                return sessionWithSolution;
+                return new SessionWithSolution(connection, scope);
             }
             catch
             {
-                sessionWithSolution.Dispose();
+                // make sure disposable objects are disposed when
+                // exceptions are thrown
```

Comment:
:bulb: This comment is unnecessary and can be removed

Reply:
this tells what this test is trying to do.